src/OptimalFilter.jl

```diff
@@ -44,7 +44,7 @@ function covariance(x::T, y::T, noise_params::NamedTuple) where T
 end
 
 # Extended covariance function /bar r(x,y), equation 8 of Dietrich and Newsam 96
-function extended_covariance(x::T, y::T, grid::NamedTuple, noise_params::NamedTuple) where T
+function extended_covariance(x::T, y::T, grid, noise_params::NamedTuple) where T
```

Comment:
It's probably better to annotate
```suggestion
function extended_covariance(x::T, y::T, grid::Grid, noise_params::NamedTuple) where T
```
here and in all other definitions below.  This is more to make sure we call the function with the right argument than else (no performance gain)